### PR TITLE
Add registration screen and API call

### PR DESCRIPTION
**Pull request by aldin-alagic**

### CODE REVIEW

**Comment by fifi98** [comment]
If I try to register with an existing email or username, I will be taken to the welcome screen with "There was a problem with your request. Please try again later. Request failed with status code 400" message. We need to inform the user that the entered email/username is taken without he needing to start the registration process all over again from the welcome screen.

**Comment by fobos531** [comment]
> If I try to register with an existing email or username, I will be taken to the welcome screen with "There was a problem with your request. Please try again later. Request failed with status code 400" message. We need to inform the user that the entered email/username is taken without he needing to start the registration process all over again from the welcome screen.

This will be added when [PR#10](https://github.com/nibble-lumen/washere-frontend/pull/10) is merged, see commit: https://github.com/nibble-lumen/washere-frontend/pull/10/commits/8c9fe9b99391ebfcaded7b406cdc52a00c38b859

File: WasHere/src/screens/Register.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        <WelcomeScreenGreen style={styles.image} />
+        <BottomSheet onClose={() => navigation.goBack()}>
+          <View style={styles.sheet}>
+            <Heading title="Create a new account" onClose={() => navigation.navigate(routes.WELCOME)} />
+            <Form
+              initialValues={{ fullname: "", username: "", email: "", password: "", passwordConfirmation: "" }}
+              onSubmit={handleSubmit}
+              validationSchema={validationSchema}>
+              <FormField autoCorrect={false} icon="person-outline" name="fullname" placeholder="Full name" />
+              <FormField autoCorrect={false} icon="desktop-outline" name="username" placeholder="Username" />
```
Comment (fobos531) [inline WasHere/src/screens/Register.js:54]: I would set `autoCapitalize` to `"none"` here as well.

File: WasHere/src/screens/Register.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+const validationSchema = Yup.object().shape({
+  fullname: Yup.string().required().label("Full name"),
+  username: Yup.string().required().min(4).label("Username"),
+  email: Yup.string().required().email().label("Email"),
+  password: Yup.string()
+    .required()
+    .matches(
+      /^.*(?=.{8,})((?=.*[!@#$%^&*()\-_=+{};:,<.>]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
+      "Password must contain at least 8 characters, one uppercase, one number and one special case character",
```
Comment (fobos531) [inline WasHere/src/screens/Register.js:26]: The word `case` is redundant here.

File: WasHere/src/screens/Register.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  passwordConfirmation: Yup.string().oneOf([Yup.ref("password"), null], "Passwords must match"),
+});
+
+const Register = ({ navigation }) => {
+  const dispatch = useDispatch();
+  const { loading } = useSelector((state) => state.auth);
+
+  const handleSubmit = ({ fullname, username, email, password }) => {
+    dispatch(register(fullname, username, email, password));
+    navigation.navigate(routes.WELCOME)
```
Comment (fobos531) [inline WasHere/src/screens/Register.js]: Is it a good idea to automatically navigate to Welcome route here? What happens if the frontend sends an API call, but the backend for whatever reason returns a 4xx response. The approach I would take here is an additional field in auth state, which keeps track of whether the registration was successful. If it was, only then should the user be navigated back to Welcome screen. See this commit for reference: https://github.com/nibble-lumen/washere-frontend/pull/10/commits/093d86a8a36a0207c663b829d5237ed995f10668

Also, when [PR#10](https://github.com/nibble-lumen/washere-frontend/pull/10) is merged, feel free to make use of `react-native-flash-message` library, as demonstrated in commit linked above.

**Review by fobos531 — CHANGES_REQUESTED** [review]

File: WasHere/src/screens/Register.js (hunk 3 above)
Comment (aldin-alagic) [reply WasHere/src/screens/Register.js]: "The approach I would take here is an additional field in auth state, which keeps track of whether the registration was successful. " That field exists and it's called apiResult.success

**Pull request merged**